### PR TITLE
Add Analyzer data store and update light importer screens

### PR DESCRIPTION
**Pull request by bogiii**
#### Proposed Changes

* Introduced Analyzer `@wordpress/data` data store. The fundament store for the future analysis methods, such as photos, typographies, etc.
* Updated light importer component screens to support showing variable data.

In the meantime, a new design brings changes to the text and adds one more screen step, but the primary purpose of this PR is the Analyzer data store, so in the following PRs, we will bring final design changes.

FYI: For presenting colors, we are providing only link colors in this dev phase. (will be updated in the following PRs)

#### Testing Instructions

* Go to `/setup/importLight?siteSlug={SLUG}`
* Provide a website (example: https://deafening-tern.jurassic.ninja/)
* Check redux actions (if you have a Redux plugin, there is a nice UI to check it)

#### Screenshot
Screenshot 2022-06-20 at 12 55 03

Related to #64774

### CODE REVIEW

File: packages/data-stores/src/analyzer/types.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	links: Color[];
+	favicon: Color[];
+	body?: Color;
+	theme_color?: Color;
+}
+
+export interface AnalyzeColorsResponse {
+	url: string;
+	status: string;
+	colors: ColorsData;
```
Comment (zaerl) [inline packages/data-stores/src/analyzer/types.ts]: I think that these types should be updated to last [url-analyzer](https://github.com/Automattic/importer-functions/blob/master/functions/url-analyzer/src/types.ts#L47-L84) types. The structure that you find in production right now has been changed. Isn't this true @jeroenpf?

**Comment by vishnugopal** [comment]
Can confirm that the state is updated fine for me!

CleanShot 2022-06-28 at 11 13 07@2x


If possible, I'd suggest making preferredPalette an array `preferredPalettes`, and just returning a single item for now. cc @bogiii @jeroenpf

**Comment by jeroenpf** [comment]
> Can confirm that the state is updated fine for me!
> 
> CleanShot 2022-06-28 at 11 13 07@2x
> 
> If possible, I'd suggest making preferredPalette an array `preferredPalettes`, and just returning a single item for now. cc @bogiii @jeroenpf

For me that's ok. This would affect @zaerl and the implementation in MC as well. I can make the change and then merge it when all are ready for it.

File: packages/data-stores/src/analyzer/types.ts (hunk 1 above)
Comment (jeroenpf) [reply packages/data-stores/src/analyzer/types.ts]: You're right. This type definition is outdated.

**Comment by zaerl** [comment]
> Can confirm that the state is updated fine for me!
> 
> CleanShot 2022-06-28 at 11 13 07@2x
> 
> If possible, I'd suggest making preferredPalette an array `preferredPalettes`, and just returning a single item for now. cc @bogiii @jeroenpf

We choose the snake case to follow the style we see on the wpcom backend. Or at least in the significant part of the endpoints we saw. Returning a single item is ok. This change will not affect production. I only need to change just one line in the last diff.

**Comment by vishnugopal** [comment]
Snake case is fine for me! Thank you! :)

**Comment by jeroenpf** [comment]
There is as PR for the response with an array of preferred palettes:  15-gh-Automattic/importer-functions

File: packages/data-stores/src/analyzer/types.ts (hunk 1 above)
Comment (bogiii) [reply packages/data-stores/src/analyzer/types.ts]: Updated.

**Pull request merged**